packages/api/amplify_api/test/dart_graphql_test.dart

```diff
@@ -375,5 +375,31 @@ void main() {
       await operation.operation.valueOrCancellation();
       expect(operation.operation.isCanceled, isTrue);
     });
+
+    test('stream should emit response with error when subscription fails',
```

Comment:
Similar comment from the utils file, test will need slight refactoring to get the response from the web socket you're looking for here.

Reply:
I think it would be tough to mock this behavior so I just left out the unit tests here and tried to make the integration test better.